Package Request: Softether Server

SoftEther VPN ("SoftEther" means "Software Ethernet") is one of the world's most powerful and easy-to-use multi-protocol VPN software.

SoftEther provides SSTP, L2TP, OpenVPN and their own specific SoftEther protocol.

SoftEther source code for ARM EABI 32-bit Linux is provided here:
https://www.softether-download.com/en.aspx?product=softether

Have a look at build notes:
https://github.com/SoftEtherVPN/SoftEtherVPN/blob/master/src/BUILD_UNIX.md

Here's a list of errors before it ends up with:
```
clang-8: error: linker command failed with exit code 1 (use -v to see invocation)
```

<details><summary><b>(Click to expand...)</b></summary>
<pre>
Preparing SoftEther VPN Server...
ranlib lib/libcharset.a
ranlib lib/libcrypto.a
ranlib lib/libedit.a
ranlib lib/libiconv.a
ranlib lib/libncurses.a
ranlib lib/libssl.a
ranlib lib/libz.a
ranlib code/vpnserver.a
gcc code/vpnserver.a -fPIE -O2 -fsigned-char -pthread -lm -lrt -Wl,--no-warn-mismatch -lpthread -L./ lib/libssl.a lib/libcrypto.a lib/libiconv.a lib/libcharset.a lib/libedit.a lib/libncurses.a lib/libz.a -ldl -o vpnserver
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: code/vpnserver.a(Network.o): in function `VPN_C53061D83562':
Network.c:(.text+0xb47c): undefined reference to `__errno_location'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: code/vpnserver.a(Network.o): in function `VPN_3E8136296102':
Network.c:(.text+0xb6dc): undefined reference to `__errno_location'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: code/vpnserver.a(Network.o): in function `VPN_FCFCDE91479E':
Network.c:(.text+0xb9a8): undefined reference to `__errno_location'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: code/vpnserver.a(Network.o): in function `VPN_15D4FF0F10FA':
Network.c:(.text+0xbb48): undefined reference to `__errno_location'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: code/vpnserver.a(Network.o): in function `VPN_4693ADC4734F':
Network.c:(.text+0xc8e0): undefined reference to `__errno_location'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: code/vpnserver.a(Network.o):Network.c:(.text+0x15224): more undefined references to `__errno_location' follow
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: code/vpnserver.a(Str.o): in function `VPN_2269FFF287AB':
Str.c:(.text+0x5e0c): undefined reference to `__ctype_b_loc'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: code/vpnserver.a(Unix.o): in function `UnixFileSize':
Unix.c:(.text+0x13e0): undefined reference to `__fxstat64'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: code/vpnserver.a(Unix.o): in function `UnixEnumDirEx':
Unix.c:(.text+0x27b8): undefined reference to `__xstat64'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: code/vpnserver.a(Internat.o): in function `IconvFreeInternal':
Internat.c:(.text+0x28a4): undefined reference to `iconv_close'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: code/vpnserver.a(Internat.o): in function `IconvStrToWideInternal':
Internat.c:(.text+0x28ec): undefined reference to `iconv_open'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: code/vpnserver.a(Internat.o): in function `IconvWideToStrInternal':
Internat.c:(.text+0x2950): undefined reference to `iconv_open'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: code/vpnserver.a(Internat.o): in function `UnixUniToStr':
Internat.c:(.text+0x2a44): undefined reference to `iconv'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: code/vpnserver.a(Internat.o): in function `UnixStrToUni':
Internat.c:(.text+0x2c50): undefined reference to `iconv'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: code/vpnserver.a(Internat.o): in function `VPN_80E915FF6F1D':
Internat.c:(.text+0x34bc): undefined reference to `iconv_close'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: code/vpnserver.a(Bridge.o): in function `VPN_49D810625981':
Bridge.c:(.text+0x2084): undefined reference to `__errno_location'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: code/vpnserver.a(Bridge.o): in function `EthGetPacketLinux':
Bridge.c:(.text+0x2238): undefined reference to `__errno_location'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: code/vpnserver.a(Console.o): in function `VPN_09E2BAB5ADA4':
Console.c:(.text+0xa34): undefined reference to `_IO_getc'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: Console.c:(.text+0xa60): undefined reference to `_IO_getc'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: Console.c:(.text+0xac0): undefined reference to `_IO_putc'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: Console.c:(.text+0xacc): undefined reference to `_IO_putc'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: Console.c:(.text+0xad8): undefined reference to `_IO_putc'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: Console.c:(.text+0xaf4): undefined reference to `_IO_putc'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: code/vpnserver.a(VLan.o): in function `VPN_40DEBE510484':
VLan.c:(.text+0x9b4): undefined reference to `__errno_location'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libssl.a(rec_layer_s3.o): in function `ssl3_write_pending':
rec_layer_s3.c:(.text+0x1020): undefined reference to `__errno_location'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libssl.a(rec_layer_s3.o): in function `ssl3_read_n':
rec_layer_s3.c:(.text+0x27a4): undefined reference to `__errno_location'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libssl.a(s3_lib.o): in function `ssl3_write':
s3_lib.c:(.text+0xce0): undefined reference to `__errno_location'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libssl.a(s3_lib.o): in function `ssl3_read_internal':
s3_lib.c:(.text+0xd48): undefined reference to `__errno_location'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libssl.a(ssl_cert.o):ssl_cert.c:(.text+0xa68): more undefined references to `__errno_location' follow
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libcrypto.a(rand_unix.o): in function `cleanup_shm':
rand_unix.c:(.text+0x194): undefined reference to `shmdt'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libcrypto.a(rand_unix.o): in function `check_random_device':
rand_unix.c:(.text+0x1c8): undefined reference to `__fxstat'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libcrypto.a(rand_unix.o): in function `rand_pool_acquire_entropy':
rand_unix.c:(.text+0x320): undefined reference to `__errno_location'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: rand_unix.c:(.text+0x480): undefined reference to `__errno_location'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: rand_unix.c:(.text+0x520): undefined reference to `__fxstat'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: rand_unix.c:(.text+0x56c): undefined reference to `shmget'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: rand_unix.c:(.text+0x58c): undefined reference to `shmat'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: rand_unix.c:(.text+0x6c8): undefined reference to `__errno_location'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: rand_unix.c:(.text+0x6fc): undefined reference to `__errno_location'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: rand_unix.c:(.text+0x764): undefined reference to `shmget'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libcrypto.a(by_dir.o): in function `get_cert_by_subject':
by_dir.c:(.text+0x1f0): undefined reference to `__xstat'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libcrypto.a(conf_lib.o): in function `OPENSSL_INIT_set_config_appname':
conf_lib.c:(.text+0xac): undefined reference to `__strdup'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libcrypto.a(conf_lib.o): in function `OPENSSL_INIT_set_config_filename':
conf_lib.c:(.text+0xe8): undefined reference to `__strdup'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libcrypto.a(conf_sap.o): in function `OPENSSL_config':
conf_sap.c:(.text+0xb4): undefined reference to `__strdup'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libcrypto.a(loader_file.o): in function `file_load':
loader_file.c:(.text+0x1710): undefined reference to `__errno_location'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: loader_file.c:(.text+0x1914): undefined reference to `__errno_location'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libcrypto.a(loader_file.o): in function `file_open':
loader_file.c:(.text+0x1b8c): undefined reference to `__xstat'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: loader_file.c:(.text+0x1b9c): undefined reference to `__errno_location'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: loader_file.c:(.text+0x1dcc): undefined reference to `__errno_location'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libcrypto.a(ui_openssl.o): in function `open_console':
ui_openssl.c:(.text+0x6c4): undefined reference to `__errno_location'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libcrypto.a(conf_def.o): in function `def_load_bio':
conf_def.c:(.text+0x130c): undefined reference to `__xstat'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: conf_def.c:(.text+0x165c): undefined reference to `__errno_location'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libedit.a(readline.o): in function `rl_completion_matches':
readline.c:(.text+0x41c): undefined reference to `__strdup'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: readline.c:(.text+0x440): undefined reference to `__strdup'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libedit.a(readline.o): in function `rl_callback_read_char':
readline.c:(.text+0x7ec): undefined reference to `__strdup'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libedit.a(readline.o): in function `username_completion_function':
readline.c:(.text+0x98c): undefined reference to `getpwent'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: readline.c:(.text+0x9c4): undefined reference to `__strdup'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: readline.c:(.text+0x9d0): undefined reference to `setpwent'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libedit.a(readline.o): in function `_rl_event_read_char':
readline.c:(.text+0x13a0): undefined reference to `__errno_location'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libedit.a(readline.o): in function `rl_callback_handler_install':
readline.c:(.text+0x17c0): undefined reference to `__strdup'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libedit.a(readline.o): in function `history_expand':
readline.c:(.text+0x2260): undefined reference to `__strdup'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: readline.c:(.text+0x2470): undefined reference to `__strdup'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: readline.c:(.text+0x29ac): undefined reference to `__strdup'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: readline.c:(.text+0x29fc): undefined reference to `__strdup'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libedit.a(readline.o):readline.c:(.text+0x2aa0): more undefined references to `__strdup' follow
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libedit.a(read.o): in function `read_char':
read.c:(.text+0x75c): undefined reference to `__errno_location'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libedit.a(tty.o): in function `tty_getty':
tty.c:(.text+0x130): undefined reference to `__errno_location'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libedit.a(tty.o): in function `T.26':
tty.c:(.text+0x2f4): undefined reference to `__errno_location'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libedit.a(fgetln.o): in function `libedit_fgetln':
fgetln.c:(.text+0xc8): undefined reference to `__errno_location'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libedit.a(tokenizer.o): in function `tok_str':
tokenizer.c:(.text+0x5a4): undefined reference to `__rawmemchr'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libedit.a(tokenizer.o): in function `tok_init':
tokenizer.c:(.text+0x640): undefined reference to `__strdup'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libedit.a(history.o): in function `history_def_del':
history.c:(.text+0x474): undefined reference to `__strdup'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libedit.a(history.o): in function `history_def_enter':
history.c:(.text+0x10d4): undefined reference to `__strdup'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libedit.a(filecomplete.o): in function `fn_complete':
filecomplete.c:(.text+0x5c0): undefined reference to `_IO_getc'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libedit.a(filecomplete.o): in function `fn_tilde_expand':
filecomplete.c:(.text+0x770): undefined reference to `__strdup'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: filecomplete.c:(.text+0x81c): undefined reference to `__strdup'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libedit.a(filecomplete.o): in function `fn_filename_completion_function':
filecomplete.c:(.text+0x9a0): undefined reference to `__strdup'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: filecomplete.c:(.text+0xb70): undefined reference to `__strdup'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libncurses.a(tty_update.o): in function `InsStr':
tty_update.c:(.text+0x9ac): undefined reference to `_IO_putc'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: tty_update.c:(.text+0xc80): undefined reference to `_IO_putc'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: tty_update.c:(.text+0xcac): undefined reference to `_IO_putc'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libncurses.a(tty_update.o): in function `PutCharLR':
tty_update.c:(.text+0x10fc): undefined reference to `_IO_putc'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: tty_update.c:(.text+0x1120): undefined reference to `_IO_putc'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libncurses.a(tty_update.o):tty_update.c:(.text+0x1130): more undefined references to `_IO_putc' follow
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libncurses.a(doalloc.o): in function `_nc_doalloc':
doalloc.c:(.text+0x3c): undefined reference to `__errno_location'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libncurses.a(lib_setup.o): in function `_nc_unicode_locale':
lib_setup.c:(.text+0x2a8): undefined reference to `nl_langinfo'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libncurses.a(lib_setup.o): in function `_nc_get_screensize':
lib_setup.c:(.text+0x3ec): undefined reference to `__errno_location'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libncurses.a(lib_setup.o): in function `_nc_setupterm':
lib_setup.c:(.text+0x5dc): undefined reference to `__strdup'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libncurses.a(lib_tputs.o): in function `_nc_outch':
lib_tputs.c:(.text+0x34): undefined reference to `_IO_putc'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libncurses.a(lib_ttyflags.o): in function `_nc_set_tty_mode':
lib_ttyflags.c:(.text+0x58): undefined reference to `__errno_location'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libncurses.a(lib_ttyflags.o): in function `_nc_get_tty_mode':
lib_ttyflags.c:(.text+0x240): undefined reference to `__errno_location'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libncurses.a(trim_sgr0.o): in function `set_attribute_9':
trim_sgr0.c:(.text+0x4c): undefined reference to `__strdup'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libncurses.a(trim_sgr0.o): in function `_nc_trim_sgr0':
trim_sgr0.c:(.text+0x2c4): undefined reference to `__strdup'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: trim_sgr0.c:(.text+0x420): undefined reference to `__strdup'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libncurses.a(access.o): in function `_nc_is_file_path':
access.c:(.text+0x30): undefined reference to `__xstat64'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libncurses.a(access.o): in function `_nc_is_dir_path':
access.c:(.text+0x74): undefined reference to `__xstat64'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libncurses.a(access.o): in function `_nc_access':
access.c:(.text+0x124): undefined reference to `__errno_location'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libncurses.a(db_iterator.o): in function `next_list_item':
db_iterator.c:(.text+0x134): undefined reference to `__strdup'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libncurses.a(lib_napms.o): in function `napms':
lib_napms.c:(.text+0x4c): undefined reference to `__errno_location'
/data/data/com.termux/files/usr/bin/arm-linux-androideabi-ld: lib/libncurses.a(lib_set_term.o): in function `delscreen':
lib_set_term.c:(.text+0x968): undefined reference to `__errno_location'
clang-8: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [Makefile:29: i_read_and_agree_the_license_agreement] Error 1
</pre>
</details>

--- a/packages/softether-vpn/CMakeLists.txt.patch
+++ b/packages/softether-vpn/CMakeLists.txt.patch
@@ -1,0 +1,13 @@
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -41,8 +41,8 @@
+ 
+   include(CheckIncludeFile)
+   Check_Include_File(sys/auxv.h HAVE_SYS_AUXV)
+-  if(EXISTS "/lib/systemd/system")
+-    set(CMAKE_INSTALL_SYSTEMD_UNITDIR "/lib/systemd/system" CACHE STRING "Where to install systemd unit files")
++  if(TRUE)
++    set(CMAKE_INSTALL_SYSTEMD_UNITDIR "@TERMUX_PREFIX@/lib/systemd/system" CACHE STRING "Where to install systemd unit files")
+   endif()
+ endif()
+ 

--- a/packages/softether-vpn/build.sh
+++ b/packages/softether-vpn/build.sh
@@ -1,0 +1,44 @@
+TERMUX_PKG_HOMEPAGE=https://www.softether.org/
+TERMUX_PKG_DESCRIPTION="An open-source cross-platform multi-protocol VPN program"
+TERMUX_PKG_LICENSE="Apache-2.0"
+TERMUX_PKG_MAINTAINER="@termux"
+TERMUX_PKG_VERSION=(5.02.5180)
+TERMUX_PKG_VERSION+=(1.0.18)
+TERMUX_PKG_SRCURL=(https://github.com/SoftEtherVPN/SoftEtherVPN/releases/download/${TERMUX_PKG_VERSION}/SoftEtherVPN-${TERMUX_PKG_VERSION}.tar.xz
+                   https://github.com/jedisct1/libsodium/archive/${TERMUX_PKG_VERSION[1]}-RELEASE.tar.gz)
+TERMUX_PKG_SHA256=(b5649a8ea3cc6477325e09e2248ef708d434ee3b2251eb8764bcfc15fb1de456
+                   b7292dd1da67a049c8e78415cd498ec138d194cfdb302e716b08d26b80fecc10)
+TERMUX_PKG_DEPENDS="libiconv, libsodium, ncurses, openssl, readline, zlib"
+TERMUX_PKG_FORCE_CMAKE=true
+TERMUX_PKG_EXTRA_CONFIGURE_ARGS="
+-DHAS_SSE2=OFF
+"
+TERMUX_PKG_HOSTBUILD=true
+TERMUX_PKG_RM_AFTER_INSTALL="lib/systemd"
+
+termux_step_post_get_source() {
+	mv libsodium-${TERMUX_PKG_VERSION[1]}-RELEASE libsodium
+}
+
+termux_step_host_build() {
+	local _PREFIX_FOR_BUILD=$TERMUX_PKG_HOSTBUILD_DIR/prefix
+	mkdir -p $_PREFIX_FOR_BUILD
+	mkdir -p libsodium
+	pushd libsodium
+	$TERMUX_PKG_SRCDIR/libsodium/configure --prefix=$_PREFIX_FOR_BUILD
+	make -j $TERMUX_MAKE_PROCESSES
+	make install
+	popd
+
+	export PKG_CONFIG_PATH=$_PREFIX_FOR_BUILD/lib/pkgconfig
+
+	termux_setup_cmake
+	cmake $TERMUX_PKG_SRCDIR
+	make -j $TERMUX_MAKE_PROCESSES
+
+	unset PKG_CONFIG_PATH
+}
+
+termux_step_post_configure() {
+	export PATH=$TERMUX_PKG_HOSTBUILD_DIR/src/hamcorebuilder:$PATH
+}

--- a/packages/softether-vpn/src-CMakeLists.txt.patch
+++ b/packages/softether-vpn/src-CMakeLists.txt.patch
@@ -1,0 +1,11 @@
+--- a/src/CMakeLists.txt
++++ b/src/CMakeLists.txt
+@@ -69,7 +69,7 @@
+ 
+   add_definitions(-DUNIX -DOS_UNIX)
+ 
+-  if(${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
++  if(${CMAKE_SYSTEM_NAME} MATCHES "Android|Linux")
+     add_definitions(-DUNIX_LINUX)
+     if("$ENV{USE_MUSL}" STREQUAL "YES")
+       add_definitions(-DUNIX_LINUX_MUSL)

--- a/packages/softether-vpn/src-Cedar-CMakeLists.txt.patch
+++ b/packages/softether-vpn/src-Cedar-CMakeLists.txt.patch
@@ -1,0 +1,10 @@
+--- a/src/Cedar/CMakeLists.txt
++++ b/src/Cedar/CMakeLists.txt
+@@ -19,7 +19,6 @@
+   RUNTIME_OUTPUT_DIRECTORY "${BUILD_DIRECTORY}"
+ )
+ 
+-cmake_host_system_information(RESULT HAS_SSE2 QUERY HAS_SSE2)
+ 
+ set(BLAKE2_SRC_PATH $<IF:$<BOOL:${HAS_SSE2}>,${TOP_DIRECTORY}/3rdparty/BLAKE2/sse,${TOP_DIRECTORY}/3rdparty/BLAKE2/ref>)
+ set(BLAKE2_SRC $<IF:$<BOOL:${HAS_SSE2}>,${BLAKE2_SRC_PATH}/blake2s.c,${BLAKE2_SRC_PATH}/blake2s-ref.c>)

--- a/packages/softether-vpn/src-Cedar-Cedar.c.patch
+++ b/packages/softether-vpn/src-Cedar-Cedar.c.patch
@@ -1,0 +1,11 @@
+--- a/src/Cedar/Cedar.c
++++ b/src/Cedar/Cedar.c
+@@ -1290,7 +1290,7 @@
+ // Initialize net service list
+ void InitNetSvcList(CEDAR *cedar)
+ {
+-	char filename[MAX_PATH] = "/etc/services";
++	char filename[MAX_PATH] = "@TERMUX_PREFIX@/etc/services";
+ 	BUF *b;
+ 	// Validate arguments
+ 	if (cedar == NULL)

--- a/packages/softether-vpn/src-Mayaqua-FileIO.c.orig.patch
+++ b/packages/softether-vpn/src-Mayaqua-FileIO.c.orig.patch
@@ -1,0 +1,13 @@
+--- a/src/Mayaqua/FileIO.c
++++ b/src/Mayaqua/FileIO.c
+@@ -20,8 +20,8 @@
+ 
+ #include <Hamcore.h>
+ 
+-static char exe_file_name[MAX_SIZE] = "/tmp/a.out";
+-static wchar_t exe_file_name_w[MAX_SIZE] = L"/tmp/a.out";
++static char exe_file_name[MAX_SIZE] = "@TERMUX_PREFIX@/tmp/a.out";
++static wchar_t exe_file_name_w[MAX_SIZE] = L"@TERMUX_PREFIX@/tmp/a.out";
+ static LIST *hamcore = NULL;
+ static HAMCORE *hamcore_io = NULL;
+ 

--- a/packages/softether-vpn/src-Mayaqua-Mayaqua.c.orig.patch
+++ b/packages/softether-vpn/src-Mayaqua-Mayaqua.c.orig.patch
@@ -1,0 +1,25 @@
+--- a/src/Mayaqua/Mayaqua.c
++++ b/src/Mayaqua/Mayaqua.c
+@@ -620,11 +620,11 @@
+ 		UINT64 now = SystemTime64();
+ 		IO *o;
+ 
+-		MakeDir("/tmp");
++		MakeDir("@TERMUX_PREFIX@/tmp");
+ 
+ 		Format(tmp2, sizeof(tmp2), "%I64u", now);
+ 
+-		Format(tmp, sizeof(tmp), "/tmp/.%s", tmp2);
++		Format(tmp, sizeof(tmp), "@TERMUX_PREFIX@/tmp/.%s", tmp2);
+ 
+ 		o = FileCreate(tmp);
+ 		if (o == NULL)
+@@ -632,7 +632,7 @@
+ 			o = FileOpen(tmp, false);
+ 			if (o == NULL)
+ 			{
+-				Print("Unable to use /tmp.\n\n");
++				Print("Unable to use @TERMUX_PREFIX@/tmp.\n\n");
+ 				exit(0);
+ 			}
+ 		}

--- a/packages/softether-vpn/src-Mayaqua-Network.c.orig.patch
+++ b/packages/softether-vpn/src-Mayaqua-Network.c.orig.patch
@@ -1,0 +1,29 @@
+--- a/src/Mayaqua/Network.c
++++ b/src/Mayaqua/Network.c
+@@ -8024,7 +8024,7 @@
+ 
+ 		GetLocalHostIP4(ip);
+ 
+-		b = ReadDump("/etc/resolv.conf");
++		b = ReadDump("@TERMUX_PREFIX@/etc/resolv.conf");
+ 		if (b != NULL)
+ 		{
+ 			char *s;
+@@ -9980,7 +9980,7 @@
+ bool UnixGetDomainName(char *name, UINT size)
+ {
+ 	bool ret = false;
+-	BUF *b = ReadDump("/etc/resolv.conf");
++	BUF *b = ReadDump("@TERMUX_PREFIX@/etc/resolv.conf");
+ 
+ 	if (b == NULL)
+ 	{
+@@ -14907,7 +14907,7 @@
+ 		return false;
+ 	}
+ 
+-	b = ReadDump("/etc/hosts");
++	b = ReadDump("@TERMUX_PREFIX@/etc/hosts");
+ 	if (b == NULL)
+ 	{
+ 		return false;

--- a/packages/softether-vpn/src-Mayaqua-Unix.c.orig.patch
+++ b/packages/softether-vpn/src-Mayaqua-Unix.c.orig.patch
@@ -1,0 +1,25 @@
+--- a/src/Mayaqua/Unix.c
++++ b/src/Mayaqua/Unix.c
+@@ -298,7 +298,7 @@
+ 	char *vm_str_list = "Hypervisor detected,VMware Virtual Platform,VMware Virtual USB,qemu,xen,paravirtualized,virtual hd,virtualhd,virtual pc,virtualpc,kvm,oracle vm,oraclevm,parallels,xvm,bochs";
+ 
+ #ifdef	UNIX_LINUX
+-	t = UnixExec("/bin/dmesg");
++	t = UnixExec("@TERMUX_PREFIX@/bin/dmesg");
+ 
+ 	if (t != NULL)
+ 	{
+@@ -363,11 +363,11 @@
+ 		// Mac OS X
+ 		if (b)
+ 		{
+-			UnixExecSilent("/usr/sbin/sysctl -w net.inet.ipsec.esp_port=4500");
++			UnixExecSilent("@TERMUX_PREFIX@/bin/sysctl -w net.inet.ipsec.esp_port=4500");
+ 		}
+ 		else
+ 		{
+-			UnixExecSilent("/usr/sbin/sysctl -w net.inet.ipsec.esp_port=4501");
++			UnixExecSilent("@TERMUX_PREFIX@/bin/sysctl -w net.inet.ipsec.esp_port=4501");
+ 		}
+ 	}
+ }

